ER-Net/ER_NET/ER_NET_tests/TcpParserTests.cs

```diff
@@ -0,0 +1,33 @@
+using System.Net;
+using ER_NET.Shared;
+using FluentAssertions;
+using Xunit;
+
+namespace ER_NET_tests
+{
+    public class TcpParserTests
+    {
+        [Fact]
+        public void Correct_TCP_received()
+        {
+            string mockMessage =
+                "ER-NET\n{\"Guid\":\"e6ad6f00-82f2-40eb-855a-54133ba29b70\",\"MessageType\":\"DiscoveryAcknowledge\"}";
+            var parser = new TcpParser();
+            Assert.Raises<TcpEventArgs>(handler => parser.OnTcpEvent += handler,
+                handler => parser.OnTcpEvent -= handler, () => { parser.OnMessageReceived(mockMessage, IPAddress.Parse("192.168.2.10")); });
+        }
+
+        [Fact]
+        public void Wrong_TCP_received()
+        {
+            string mockMessage =
+                "\n{\"Guid\":\"e6ad6f00-82f2-40eb-855a-54133ba29b70\",\"MessageType\":\"DiscoveryAcknowledge\"}";
+            var parser = new TcpParser();
+            using (var monitor = parser.Monitor())
+            {
+                parser.OnMessageReceived(mockMessage, IPAddress.Parse("192.168.2.10"));
+                monitor.Should().NotRaise("OnTcpEvent");
+            }
+        }
+    }
+}
```

Comment:
ik heb alles doorgelezen en het ziet er allemaal erg prima uit. er staat niet veel commentaar bij je code, maar de namen van functies en variabelen zijn duidelijk en dan is commentaar wat mij betreft niet nodig.

Reply:
top